broker/src/main/java/io/zeebe/broker/system/partitions/impl/PartitionTransitionImpl.java

```diff
@@ -18,101 +19,78 @@
 import java.util.List;
 import org.slf4j.Logger;
 
-public class PartitionTransitionImpl<T> implements PartitionTransition {
+public class PartitionTransitionImpl implements PartitionTransition {
 
   private static final Logger LOG = Loggers.SYSTEM_LOGGER;
+  private static final List<PartitionStep> EMPTY_LIST = ImmutableList.of();
```

Comment:
For an empty list, we can use the ones from the standard Java lib instead of an external lib.

```suggestion
  private static final List<PartitionStep> EMPTY_LIST = List.of();
```

Or,

```suggestion
  private static final List<PartitionStep> EMPTY_LIST = Collections.emptyList();
```

Reply:
The idea was that this should never contain anything and an ImmutableList would enforce that. We already use it a lot in the project so it's not adding a new dependency.

Reply:
Ok. Just as a side note. Both suggestions from the JDK create an empty list that is also immutable :)

Reply:
Ah my bad, I didn't know that. I'll fix it then, thanks!